src/hotspot/share/oops/constantPool.inline.hpp

```diff
@@ -30,13 +30,13 @@
 #include "oops/cpCache.inline.hpp"
 #include "runtime/atomic.hpp"
 
-inline CPSlot ConstantPool::slot_at(int which) const {
+inline Symbol* ConstantPool::slot_at(int which) const {
   assert(is_within_bounds(which), "index out of bounds");
   assert(!tag_at(which).is_unresolved_klass() && !tag_at(which).is_unresolved_klass_in_error(), "Corrupted constant pool");
   // Uses volatile because the klass slot changes without a lock.
   intptr_t adr = Atomic::load_acquire(obj_at_addr(which));
   assert(adr != 0 || which == 0, "cp entry for klass should not be zero");
-  return CPSlot(adr);
+  return (Symbol*)adr;
```

Comment:
I think you can get rid of all slot_at* and replace them with symbol_at().  Since the symbols don't change after the constant pool is initialized, all this volatile isn't needed.  Used to be used to replace Symbol* with Klass* also.  @iklam can confirm.

Reply:
`ConstantPool` objects are no modified after the class is linked, and linking is done single threaded. So there's no need to use the `Atomic::load_acquire` in the above code. I agree with Coleen that the above function can be replaced with symbol_at().